### PR TITLE
[INFLIB] Fix INFCONTEXT structure to be compatible with the official definition

### PR DESCRIPTION
**Pull request by tkreuzer**
This makes inflib work on x64 and usetup crash later.

### CODE REVIEW

File: sdk/lib/inflib/infput.c [hunk 1]
```diff
@@ -246,13 +253,16 @@ InfpAddLineWithKey(PINFCONTEXT Context, PCWSTR Key)
...
+
+  if (NULL == Context)
     {
       DPRINT1("Invalid parameter\n");
       return INF_STATUS_INVALID_PARAMETER;
     }
 
-  if (NULL == InfpAddFieldToLine(Context->Line, Data))
+  Line = InfpGetLineForContext(Context);
+  if (NULL == InfpAddFieldToLine(Line, Data))
```
Comment (HBelusca) [inline sdk/lib/inflib/infput.c]: Indentation fixes please throughout everywhere!

File: sdk/lib/inflib/infget.c [hunk 2]
```diff
@@ -360,10 +364,12 @@ InfpGetLineCount(HINF InfHandle,
 LONG
 InfpGetFieldCount(PINFCONTEXT Context)
 {
-  if (Context == NULL || Context->Line == NULL)
-    return 0;
+  PINFCACHELINE Line;
 
-  return ((PINFCACHELINE)Context->Line)->FieldCount;
+  Line = InfpGetLineForContext(Context);
```
Comment (HBelusca) [inline sdk/lib/inflib/infget.c]: You may keep checking for Context == NULL.

File: sdk/lib/inflib/infcore.c [hunk 3]
```diff
@@ -249,6 +251,74 @@ InfpAddLine(PINFCACHESECTION Section)
   return Line;
 }
 
+PINFCACHESECTION
+InfpFindSectionById(PINFCACHE Cache, UINT Id)
+{
+    PINFCACHESECTION Section;
+
+    Section = Cache->FirstSection;
```
Comment (HBelusca) [inline sdk/lib/inflib/infcore.c]: for-loop ?

File: sdk/lib/inflib/infcore.c [hunk 4]
```diff
@@ -249,6 +251,74 @@ InfpAddLine(PINFCACHESECTION Section)
...
+
+    return InfpFindSectionById(Cache, Context->Section);
+}
+
+PINFCACHELINE
+InfpFindLineById(PINFCACHESECTION Section, UINT Id)
+{
+    PINFCACHELINE Line;
+
+    Line = Section->FirstLine;
```
Comment (HBelusca) [inline sdk/lib/inflib/infcore.c]: for-loop ?

File: sdk/lib/inflib/infget.c (hunk 2 above)
Comment (tkreuzer) [reply sdk/lib/inflib/infget.c]: That's done through InfpGetLineForContext already, no need to replicate the same check throughout all levels.

File: sdk/lib/inflib/infcore.c (hunk 3 above)
Comment (tkreuzer) [reply sdk/lib/inflib/infcore.c]: Done

File: sdk/lib/inflib/infcore.c (hunk 4 above)
Comment (tkreuzer) [reply sdk/lib/inflib/infcore.c]: done

File: sdk/lib/inflib/infput.c [hunk 5]
```diff
@@ -201,40 +202,46 @@ InfpFindOrAddSection(PINFCACHE Cache,
     }
 
   (*Context)->Inf = Cache;
-  (*Context)->Section = InfpFindSection(Cache, Section);
-  (*Context)->Line = NULL;
-  if (NULL == (*Context)->Section)
+  CacheSection = InfpFindSection(Cache, Section);
+  (*Context)->Line = 0;
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply sdk/lib/inflib/infput.c]: done

**Pull request merged**